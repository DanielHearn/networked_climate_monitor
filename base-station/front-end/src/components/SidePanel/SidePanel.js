export default {
  name: 'side-panel',
  props: {},
  data: function() {
    return {}
  },
  watch: {},
  methods: {}
}
