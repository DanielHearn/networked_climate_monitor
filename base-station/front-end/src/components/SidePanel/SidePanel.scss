@import "./../../scss/variables.scss";

.side-panel {
  width: 22em;
  padding: 1em;
  background: $dark;
  color: $white;
  z-index: 1;
  height: 100%;
  display: flex;
  flex: 1 auto;
  flex-direction: column;
  .side-panel-content {
    display: flex;
    flex-direction: column;
    height: 100%;
    .sub-heading:first-child {
      margin-top: 0;
    }
  }
}