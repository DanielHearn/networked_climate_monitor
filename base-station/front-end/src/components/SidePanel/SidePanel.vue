<template>
  <div class="side-panel" :class="{ mobile: mobile, active: open }">
    <div v-if="mobile" class="mobile-header">
      <p class="sub-heading">{{ title }}</p>
      <button @click="toggleOpen">
        <i v-if="open" class="material-icons">keyboard_arrow_up</i>
        <i v-else class="material-icons">keyboard_arrow_down</i>
      </button>
    </div>
    <div
      class="side-panel-content"
      :class="{ hidden: !open && this.$store.state.mobile }"
    >
      <slot></slot>
    </div>
  </div>
</template>

<script src="./SidePanel.js"></script>

<style lang="scss" src="./SidePanel.scss"></style>
