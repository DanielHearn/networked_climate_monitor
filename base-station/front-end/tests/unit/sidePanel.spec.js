import { mount } from '@vue/test-utils'
import SidePanel from './../../src/components/SidePanel/SidePanel.vue'

describe('SidePanel.vue', () => {
  it('Header Slot', () => {
    const content = '<p>Header Content</p>'
    const wrapper = mount(SidePanel, {
      slots: {
        header: content
      }
    })

    expect(wrapper.html()).toContain(content)
  })

  it('Main Slot', () => {
    const content = '<p>Main Content</p>'
    const wrapper = mount(SidePanel, {
      slots: {
        content: content
      }
    })

    expect(wrapper.html()).toContain(content)
  })

  it('Main and Content Slots', () => {
    const mainContent = '<p>Main Content</p>'
    const headerContent = '<p>Header Content</p>'
    const wrapper = mount(SidePanel, {
      slots: {
        content: mainContent,
        header: headerContent
      }
    })

    expect(wrapper.html()).toContain(mainContent)
    expect(wrapper.html()).toContain(headerContent)
  })
})
