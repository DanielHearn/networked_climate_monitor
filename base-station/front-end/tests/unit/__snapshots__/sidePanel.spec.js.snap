// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel.vue has the expected html structure 1`] = `
<div
  class="side-panel"
>
  <div
    class="side-panel-header"
  >
    <p>
      Header Content
    </p>
  </div>
   
  <div
    class="side-panel-content"
  >
    <p>
      Main Content
    </p>
  </div>
</div>
`;
